Animations/vendor/CMakeLists.txt

```diff
@@ -0,0 +1,104 @@
+# Vendored libraries
+
+# This file acts as a proxy between all the vendored libraries
+# and the CMake build system.
+
+###############
+# CMake setup #
+###############
+include(ExternalProject)
+
+############
+# cppUtils #
+############
+add_library(CppUtils INTERFACE)
+target_include_directories(CppUtils INTERFACE "${CMAKE_CURRENT_LIST_DIR}/cppUtils/single_include")
+
+########
+# GLFW #
+########
+set(GLFW_BUILD_DOCS OFF CACHE BOOL "" FORCE)
+set(GLFW_BUILD_TESTS OFF CACHE BOOL "" FORCE)
+set(GLFW_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
+add_subdirectory(GLFW)
+
+############
+# Freetype #
+############
+add_subdirectory(freetype)
+
+##########
+# OpenAL #
+##########
+set(ALSOFT_UTILS OFF)
+set(ALSOFT_NO_CONFIG_UTIL ON)
+set(ALSOFT_EXAMPLES OFF)
+set(ALSOFT_INSTALL OFF)
+set(ALSOFT_INSTALL_CONFIG OFF)
+set(ALSOFT_INSTALL_HRTF_DATA OFF)
+set(ALSOFT_INSTALL_AMBDEC_PRESETS OFF)
+set(ALSOFT_INSTALL_EXAMPLES OFF)
+set(ALSOFT_INSTALL_UTILS OFF)
+add_subdirectory(openal)
+
+########
+# Luau #
+########
+set(CMAKE_CXX_STANDARD 17) # Luau doesn't build with C++20
+
+set(LUAU_BUILD_CLI OFF)
+set(LUAU_BUILD_TESTS OFF)
+add_subdirectory(luau)
+
+set(CMAKE_CXX_STANDARD 20) # See above
+
+#############
+# Onigurama #
+#############
+add_subdirectory(onigurama)
+
+############
+# TinyXML2 #
+############
+add_subdirectory(tinyxml2)
+
+###########
+# PlutoVG #
+###########
+add_subdirectory(plutovg)
+
+#############
+# dearimgui #
+#############
+include(DearImGUI.cmake)
+
+##########
+# ffmpeg #
+##########
+include(ffmpeg.cmake)
+
+########
+# Glad #
+########
+add_subdirectory(glad)
+
+#######
+# GLM #
+#######
+add_subdirectory(glm)
+
+####################
+# NativeFileDialog #
+####################
+include(NativeFileDialog.cmake)
+
+############
+# nlohmann #
+############
+add_subdirectory(nlohmann)
```

Comment:
`nlohmann` is a single include library, is there any advantage to adding it here instead of just adding the include directory?

Reply:
`nlohmann`  is getting actively changed upstream, sometimes also the directory layout and build options change. Using it via CMake keeps compatibility with upstream, as we don't assume anything about the library directory layout. Generally it is a good idea to treat CMake packages as blackboxes with their only API being the common CMake settings and options declared by the library.

This helps us to more easily enable future dependency upgrades.